### PR TITLE
Upgrade EVMC

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: src/hera.cpp [hunk 1]
```diff
@@ -54,7 +54,7 @@ struct hera_instance : evmc_instance {
   bool use_evm2wasm_js_trace = false;
 #endif
 
-  hera_instance() : evmc_instance({EVMC_ABI_VERSION, nullptr, nullptr, nullptr}) {}
+  hera_instance() : evmc_instance({EVMC_ABI_VERSION, "hera", "0.0.0", nullptr, nullptr, nullptr}) {}
```
Comment (axic) [inline src/hera.cpp]: So actually cable would be useful here to have the buildinfo struct: )

**Comment by axic** [comment]
Fails building:
```
/home/builder/cpp-ethereum/hera/src/eei.h:29:10: fatal error: 'evmc/evmc.h' file not found
#include <evmc/evmc.h>
         ^~~~~~~~~~~~~
```

**Comment by jakelang** [comment]
should just be `<evmc.h>` since it is already a subdirectory according to cmake

**Comment by chfast** [comment]
I just have to upgrade EVMC in cpp-ethereum as well to test Hera.

**Comment by chfast** [comment]
cpp-ethereum upgraded as well. https://github.com/ethereum/cpp-ethereum/pull/4976
This is a circular dependency, but we should be fine.

@jakelang I moved the evmc header from `include/evmc.h` to `include/evmc/evmc.h`. In other projects having an include prefix like `#include <project-name/header.h` works better. Here we have single header so this is not crucial, but still I try to avoid having very different project files layout if possible.

**Comment by chfast** [comment]
Tests passed.

**Pull request merged**